### PR TITLE
Stablize GCP Tests

### PR DESCRIPTION
**Pull request by robmorgan**
This PR continues on from https://github.com/gruntwork-io/terratest/pull/756 and aims to fix a consistent test failure in `TestPackerBasicExample`. We have broken the GCP tests out to a seperate Docker executor for better isolation and resiliency. Using the Docker executor ensures GCP tests do not errornously make metadata network calls within CircleCI's private environment. We also bump Packer to v1.6.6 as it includes improved error handling in the GCP builder.

### Tasks

- [x] Split GCP test execution out to a separate CircleCI docker executor.
- [x] Tag all GCP tests as `gcp`.

### Summary of Changes to GCP Account

- [X] Ensure the default network allows inbound SSH connections.
- [X] Remove dangling TICK firewall rules
- [x] Cleanup temporary SSH keys

### CODE REVIEW

**Comment by robmorgan** [comment]
Looks like the tests passed in https://app.circleci.com/pipelines/github/gruntwork-io/terratest/377/workflows/fbb4a896-9450-4d45-ad8c-a8daf888ba95. They were still a bit flakey, but I think now that I've cleaned up some of the SSH keys, this PR should be good to go.

File: .circleci/config.yml [hunk 1]
```diff
@@ -140,6 +136,59 @@ jobs:
...
+      - attach_workspace:
+          at: /home/circleci
+
+      # install the google cloud sdk
+      - run:
+          command: |
+            echo 'deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main' | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+            sudo apt-get install -y apt-transport-https ca-certificates gnupg
+            curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo apt-key --keyring /usr/share/keyrings/cloud.google.gpg add -
+            sudo apt-get update && sudo apt-get install -y google-cloud-sdk
```
Comment (robmorgan) [inline .circleci/config.yml:154]: I'm wondering if I could add a new helper to `module-ci` and reference it here to install the Google Cloud SDK.

Comment (brikis98) [reply .circleci/config.yml:154]: Is this something we need in every GCP repo? If so, adding it to `module-ci` makes sense.

File: .circleci/config.yml [hunk 2]
```diff
@@ -140,6 +136,59 @@ jobs:
       - store_test_results:
           path: /tmp/logs
 
+  gcp_test:
```
Comment (brikis98) [inline .circleci/config.yml:139]: NIT: add comment that the GCP tests are intentionally in a separate Docker based build, and add a link to this very PR. That way, future maintainers won't be tempted to change it back.

**Review by brikis98 — APPROVED** [review]
LGTM, thanks Rob!

File: .circleci/config.yml (hunk 1 above)
Comment (robmorgan) [reply .circleci/config.yml:154]: @brikis98 yes, we'll ultimately adopt whatever logic I figure out here for all the GCP repos. I'll move it to `module-ci` then.

**Review by infraredgirl — DISMISSED** [review]
LGTM!

File: .circleci/config.yml (hunk 2 above)
Comment (robmorgan) [reply .circleci/config.yml:156]: Added a comment block.

File: .circleci/config.yml (hunk 1 above)
Comment (robmorgan) [reply .circleci/config.yml]: Proposing a `gcp-helper` module in: https://github.com/gruntwork-io/terraform-aws-ci/pull/264

**Comment by robmorgan** [comment]
Looks like the tests passed in https://app.circleci.com/pipelines/github/gruntwork-io/terratest/384/workflows/3643c443-6969-45be-bace-4f7b6153ff20

**Review by infraredgirl — APPROVED** [review]

**Pull request merged**